ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -91,69 +159,396 @@ def loop(self):
+                        self.f_wpdeq[l_ptr].waypoint.twist.twist.linear.x = last_wpd.V
+                        self.f_wpdeq[l_ptr].V = last_wpd.V
+                        self.state = 'maintainspeed'
+                    else:
+                        if self.f_wpdeq[l_ptr].V < 0.25:
+                            self.f_wpdeq[l_ptr].waypoint.twist.twist.linear.x = 0.0
+                        else:
+                            self.f_wpdeq[l_ptr].waypoint.twist.twist.linear.x = \
+                                min(self.f_wpdeq[l_ptr].V, self.f_wpdeq[l_ptr].max_V)
+                        # end for
+                    # end if else
+                # end if
+
+                if self.state == 'JMTspeedup':
+                    wp_check = self.JMTD_at(disp, self.JMTcoeffs, self.JMT_time,
+                                            self.JMT_max_time, l_ptr)
+                    if wp_check == -1:
+                        self.f_wpdeq[l_ptr].waypoint.twist.twist.linear.x = \
+                            min(last_wpd.V, self.f_wpdeq[l_ptr].max_V)
+                        self.state = 'maintainspeed'  # Assume gone past end
+                        self.f_wpdeq[l_ptr].state = self.state
+                        self.JMTcoeffs = None
+                        self.JMT_start_ptr = self.f_wpdeq[l_ptr].ptr_id
+                        self.JMT_time = 0.0
+                        self.JMT_max_time = 0.0
+
+                    else:
+                        self.f_wpdeq[l_ptr].waypoint.twist.twist.linear.x = \
+                            min(self.f_wpdeq[l_ptr].V, self.f_wpdeq[l_ptr].max_V)
+
+                if self.state == 'startspeedup':
+                    # dist = self.distance(last_wpd.ptr_id, self.f_wpdeq[l_ptr].ptr_id)
+                    self.f_wpdeq[l_ptr].A = self.default_accel
+                    v = max(0.5 , math.sqrt(self.f_wpdeq[l_ptr].A * disp * 2.0))
+                    v = min(v, self.f_wpdeq[l_ptr].max_V)
+                    self.f_wpdeq[l_ptr].V = v
+                    self.f_wpdeq[l_ptr].waypoint.twist.twist.linear.x = v
+
+                last_wpd = self.f_wpdeq[l_ptr]
+                rospy.loginfo("state = {}, ptr= {}, v= {}, a= {}, dist= {}".
+                            format(self.state, last_wpd.ptr_id, last_wpd.V,
+                                    last_wpd.A, disp))
             else:
-                self.default_velocity = config['dyn_default_velocity']
+                disp = self.distance(self.f_wpdeq[l_ptr].phase_start_ptr, self.f_wpdeq[l_ptr].ptr_id)
+                last_wpd = self.f_wpdeq[l_ptr]
+                rospy.loginfo("skipping recalculation "\
+                              "state = {}, ptr= {}, v= {}, a= {}, dist= {}".
+                              format(self.state, last_wpd.ptr_id, last_wpd.V,
+                                     last_wpd.A, disp))
+                
+
+    # check for state transitions
+    def state_spinner(self, last_dwp_ptr, dwp_ptr):
```

Comment:
last_dwp_ptr refers to the previous waypoint and dwp_ptr is the current waypoint, right?  If so, it may be better to change the naming from "last" to "prev" so it's not confused with the last waypoint in the deque or something.